src/modules/rating/rating.ts

```diff
@@ -0,0 +1,45 @@
+import { ObjectId } from 'mongodb';
+import { useCollection } from 'db';
+import type { RatingForm, Rating, Name } from 'prytaneum-typings';
+
+import createHttpError from 'http-errors';
+
+const addRating = async (
+    rating: RatingForm,
+    townhallId: string,
+    userId?: string,
+    userName?: Name
+): Promise<void> => {
+    const ratingDoc: Pick<Rating<ObjectId>, 'meta' | 'ratings' | 'feedback'> = {
+        meta: {
+            townhallId: new ObjectId(townhallId),
+            createdAt: new Date(),
+            updatedAt: new Date(),
+        },
+        ratings: rating.values,
+        feedback: rating.feedback,
+    };
+    if (userId && userName) {
+        ratingDoc.meta.createdBy = { _id: userId, name: userName };
+        ratingDoc.meta.updatedBy = { _id: userId, name: userName };
+    }
+    const doc = await useCollection('Ratings', (Ratings) => {
+        return Ratings.insertOne(ratingDoc);
+    });
+    if (!doc) throw createHttpError(404, 'Error when adding rating');
```

Comment:
A 404 is a not found error. This should be a 500 since this is an insert and there's no schema enforcement (something is probably wrong with the connection or the database itself) -- there's nothing a user can do to fix it.